pkg/registry/apis/provisioning/register.go

```diff
@@ -42,33 +42,46 @@ var (
 
 // This is used just so wire has something unique to return
 type ProvisioningAPIBuilder struct {
-	cfg               *setting.Cfg
+	urlProvider       func(namespace string) string
+	webhookSecreteKey string
+
 	getter            rest.Getter
 	localFileResolver *LocalFolderResolver
 	logger            *slog.Logger
 }
 
-func NewProvisioningAPIBuilder(cfg *setting.Cfg, local *LocalFolderResolver) *ProvisioningAPIBuilder {
+// This constructor will be called when building a multi-tenant apiserveer
+// Avoid adding anything that secretly requires additional hidden dependencies
+// like *settings.Cfg or core grafana services that depend on database connections
```

Comment:
for reference... .this will eventually need to be called from:
https://github.com/grafana/grafana-enterprise/blob/main/src/pkg/extensions/apiserver/factory.go#L280